src/main/java/com/dmdirc/ui/messages/EventFormatter.java

```diff
@@ -155,4 +155,15 @@ private String applyDisplayProperties(final DisplayPropertyMap displayProperties
         return res.toString();
     }
 
+    /**
+     * Get the displayable properties map for this event according to the formatter.
+     *
+     * @param event Event to get properties for.
+     * @return DisplayPropertyMap from formatter.
+     */
+    public DisplayPropertyMap getFormatDisplayableProperties(final DisplayableEvent event) {
+        return formatProvider.getFormat(event.getClass()).isPresent()
+                ? formatProvider.getFormat(event.getClass()).get().getDisplayProperties()
```

Comment:
It's the format.yml properties I wanted to expose really (as you can already get the event ones with event.getDisplayProperties()) - but I wanted to be sure this always returned *something* I guess I could return an empty  DisplayPropertyMap?

`formatProvider.getFormat(event.getClass()).orElse(DisplayPropertyMap.EMPTY).getDisplayProperties()` or so?